src/api/auth/api.ts

```diff
@@ -0,0 +1,10 @@
+
+import publicApi from "src/configs/publicApi";
+import type { IAuthResponse } from "./types";
+
+export const getAuthToken = async (): Promise<IAuthResponse> => {
+  return publicApi
+    .get("/auth/anonymous?platform=subscriptions")
+    .then((res) => res.data);
```

Comment:
you should use `async/await` syntax with `try catch` construction or `fetch` with `than catch` construction when you work with Promises.

Reply:
Fortunately, react-query does this for me